api-server/src/venues/venues.controller.ts

```diff
@@ -69,6 +72,21 @@ export class VenuesController {
             })
             .then(venue => new SingleVenueResponse({ venue }));
     }
+
+    @Post('/get-gps-from-google-maps-link')
+    @ApiOperation({summary: 'get gps coordinates from a venue google maps link'})
+    @ApiResponse({
+        status: 200,
+        description: 'gps from google maps links',
+        type: GetGPSCoordinatesResponse
+    })
+    getGpsFromGoogleMapsLink(@Body() req: GetGPSCoordinatesRequest): Promise<GetGPSCoordinatesResponse> {
+        const { googleMapsLink } = req;
+
+        return this.gpsService.getCoordinatesFromGoogleMapLink(googleMapsLink).then((gpsCoordinates) => new GetGPSCoordinatesResponse({
+            gpsCoordinates
+        }))
+    }
```

Comment:
Maybe I'm missing something, but why is this a `post`? Doesn't appear to create or change anything server-side.

Although I wonder if it would make sense (and if Nest even supports this...) to support both `get` and `put` for this endpoint; `get` just queries and returns the coordinates, `put` queries and updates the venue with the coordinates without another round-trip to the client.

Reply:
I went this route mouse because I was thining queryString felt weird given the url is required, but I  also didn't want to mess with making sure a url was properly serialized as a path param. PostBody felt like the best place to put it, but you're right probably makes more sense as a get